src/Amazon.CloudWatch.EMF/Model/MetricsContext.cs

```diff
@@ -14,6 +15,11 @@ public class MetricsContext
         /// </summary>
         private readonly MetricDirective _metricDirective;
 
+        /// <summary>
+        /// Holds the metric key and its resolution type for validation checks
+        /// </summary>
+        private readonly ConcurrentDictionary<string, StorageResolution> _storageResolutionMetrics = new ConcurrentDictionary<string, StorageResolution>();
```

Comment:
Nit: Better name would be `metricNameAndResolutionMap` instead of `storageResolutionMetric`

Reply:
Done